drizzle-zod/src/mysql/index.ts

```diff
@@ -0,0 +1,243 @@
+	TNoOptional extends true ? TType
+		: GetColumnConfig<TColumn, 'hasDefault'> extends true ? z.ZodOptional<TType>
+		: TType;
+
+type GetZodType<TColumn extends AnyPgColumn> = GetColumnConfig<TColumn, 'data'> extends infer TType
+	? TColumn extends PgEnumColumn<PgEnumColumnConfig & { values: infer TValues extends [string, ...string[]] }>
+		? z.ZodEnum<TValues>
+	: TType extends number ? z.ZodNumber
+	: TType extends string ? z.ZodString
+	: TType extends boolean ? z.ZodBoolean
+	: TType extends Date ? z.ZodDate
+	: z.ZodAny
+	: never;
+
+type ConvertKeyName<K extends string, TCase extends 'snake' | 'camel' | undefined> = undefined extends TCase ? K
+	: 'snake' extends TCase ? ToSnakeCase<K>
+	: SnakeToCamelCase<K>;
+
+export type BuildInsertSchema<
+	TTable extends AnyPgTable,
+	TCase extends 'snake' | 'camel' | undefined,
+	TRefine extends Record<string, z.ZodTypeAny | undefined>,
+	TNoOptional extends boolean = false,
+> = GetTableConfig<TTable, 'columns'> extends infer TColumns extends Record<string, AnyPgColumn> ? {
+		[K in keyof TColumns & string as ConvertKeyName<K, TCase>]: MaybeOptional<
+			TColumns[K],
+			ConvertKeyName<K, TCase> extends keyof TRefine
+				? TRefine[ConvertKeyName<K, TCase>] extends z.ZodTypeAny ? TRefine[ConvertKeyName<K, TCase>]
+				: GetZodType<TColumns[K]>
+				: GetZodType<TColumns[K]>,
+			TNoOptional
+		>;
+	}
+	: never;
+
+export type GetRequiredConfigFields<T extends AnyPgTable> = GetTableConfig<T, 'columns'> extends
+	infer TColumns extends Record<string, AnyPgColumn> ? {
+		[K in keyof TColumns]: TColumns[K] extends PgText<PgTextConfig & { values: infer TValues }>
+			? [string, ...string[]] extends TValues ? never : TValues
+			: never;
+	}
+	: never;
+
+export type RequiredFieldsConfig<
+	TTable extends AnyPgTable,
+> = Simplify<
+	{
+		[K in keyof GetRequiredConfigFields<TTable> as [GetRequiredConfigFields<TTable>[K]] extends [never] ? never : K]: {
+			anyString: true;
+		} | { enum: GetRequiredConfigFields<TTable>[K] };
+	}
+>;
+
+export type Refine<
+	TTable extends AnyPgTable,
+	TCase extends 'snake' | 'camel' | undefined,
+	TRefine extends {
+		[K in keyof BuildInsertSchema<TTable, TCase, {}>]?: z.ZodTypeAny;
+	},
+> =
+	| { [K in keyof TRefine]: K extends keyof BuildInsertSchema<TTable, TCase, {}> ? TRefine[K] : never }
+	| ((
+		fields: BuildInsertSchema<TTable, TCase, {}, true>,
+	) => { [K in keyof TRefine]: K extends keyof BuildInsertSchema<TTable, TCase, {}> ? TRefine[K] : never });
+
+export function createInsertSchema<TTable extends AnyPgTable>(
+	table: TTable,
+): z.ZodObject<BuildInsertSchema<TTable, undefined, {}>>;
+export function createInsertSchema<
+	TTable extends AnyPgTable,
+	TCase extends 'snake' | 'camel',
+>(
+	table: TTable,
+	/**
+	 * @param convertToCase Convert keys to snake_case or camelCase
+	 */
+	convertToCase: TCase,
+): z.ZodObject<BuildInsertSchema<TTable, TCase, {}>>;
+export function createInsertSchema<
+	TTable extends AnyPgTable,
```

Comment:
`AnyPgTable` is used for postgres specific functions. We need to use `AnyMySqlTable` here